### ISSUE TITLE
Make a ball never lose energy on bounce? I.e Pong

### ISSUE THREAD

**Issue by wilderbonsai** [donmccurdy/aframe-physics-system#80]
Hey there! First off, awesome, easy to use add on for a-frame! Enjoying having this very much.

Question though, I can't seem to get the physics to a point where a ball will bounce forever. I.e a pong type game where a ball will have a constant velocity the whole time no matter how much it bounces.

I've put gravity as 0 and tried changing all the given params (friction, restitution, mass, damping ect), but the ball seems to always lose energy at every bounce. Am I missing something? Is this possible?

I considered adding an impulse at point of collision, but then it seems I'd have to figure out the vector of the resulting collision and do the math myself, which wouldn't be my strong suit...

Here is a codepen, of a ball between two planes with gravity:0, and friction and restitution set > 1.
https://codepen.io/wilderbonsai/pen/PQpreE

Thanks advanced for the help!

**Comment by anguyen09** [comment]
Mm I think I figured it out, after diving into Cannon.js for a while.

It seems like a-frame physics was not putting a Material on the Body of the entities, therefore the the defaultContactMaterial was always being used, and the friction/restitution parameters were not being considered. Once I manually made Materials and a new Contact Material in JS, it worked.

`var mat1 = new CANNON.Material();

var contact1 = new CANNON.ContactMaterial(mat1, mat1, { friction: 1, restitution: 1 });

ball.addEventListener('body-loaded', function () {
  ball.body.world.addContactMaterial(contact1);
  ball.body.material = mat1;
});

.... then same for all entities that I want the physics for
wall.body.material = mat1 
wall2.body.material = mat1 ect
`
And then the same for any entity that I wanted with the new physics. So wall.body.material = mat1 ect.

I couldn't find any way to set the Material from the docs of this library, so maybe this is a bug afterall? That materials weren't being set initially for bodies? Maybe I'll dive into this code and check it out and put a PR in if I have time.
